adagrams/game.py

```diff
@@ -1,11 +1,156 @@
+from operator import le
+import random
+LETTER_POOL = {
+    'A': 9, 
+    'B': 2, 
+    'C': 2, 
+    'D': 4, 
+    'E': 12, 
+    'F': 2, 
+    'G': 3, 
+    'H': 2, 
+    'I': 9, 
+    'J': 1, 
+    'K': 1, 
+    'L': 4, 
+    'M': 2, 
+    'N': 6, 
+    'O': 8, 
+    'P': 2, 
+    'Q': 1, 
+    'R': 6, 
+    'S': 4, 
+    'T': 6, 
+    'U': 4, 
+    'V': 2, 
+    'W': 2, 
+    'X': 1, 
+    'Y': 2, 
+    'Z': 1
+}
+SCORE_CHART =  {
+    'A': 1, 
+    'B': 3, 
+    'C': 3, 
+    'D': 2, 
+    'E': 1, 
+    'F': 4, 
+    'G': 2, 
+    'H': 4, 
+    'I': 1, 
+    'J': 8, 
+    'K': 5, 
+    'L': 1, 
+    'M': 3, 
+    'N': 1, 
+    'O': 1, 
+    'P': 3, 
+    'Q': 10, 
+    'R': 1, 
+    'S': 1, 
+    'T': 1, 
+    'U': 1, 
+    'V': 4, 
+    'W': 4, 
+    'X': 8, 
+    'Y': 4, 
+    'Z': 10
+}
+def create_bag_of_letters():
+    """
+    Returns an array containing all the letters of the "LETTER_POOL" dictionary 
+    according to the frequency associated with each letter.
+    Ex: Since the value of 'A' is 9, the first 9 elements of this list will be 'A' 
+    appearing 9 times, and so on.
+    bag_of_letters = ['A', 'A', ..., 'B', ...]
+    """
+    bag_of_letters = []
+    for letter in LETTER_POOL:
+        number_of_letters = LETTER_POOL[letter]
+        for i in range(number_of_letters):
+            bag_of_letters.append(letter)
+    return bag_of_letters
+
 def draw_letters():
-    pass
+    hand_of_letters = [] # to be filled with 10 one-letter strings
+    bag_of_letters = create_bag_of_letters()
+   
+    for i in range(10):
+        #size_of_letter_bag = len(bag_of_letters)
+        letter_picked = random.choice(bag_of_letters) 
+        hand_of_letters.append(letter_picked)
+        bag_of_letters.remove(letter_picked)
+
+    return hand_of_letters
 
 def uses_available_letters(word, letter_bank):
-    pass
+    copy_of_letter_bank = list(letter_bank)
+    upper_case_word = word.upper()
+    #We checked to make sure that removing wouldn't affect original letter_bank array here:
+    # copy_of_letter_bank.remove('B')
+    # print("letter bank:", letter_bank)
+    #print("copy of letter bank:" , copy_of_letter_bank)
```

Comment:
Minor style preference: Much as I said in the last function, there's not as much benefit to keeping this type of "exploratory" code in, even while commented after y'all figure it out. That being said, writing a bit of code like this to figure things out is a really good tendency to have! Just, don't keep it in. :3

Reply:
There are some other places with commented code, but from here on out, I'll skip pointing it out to not repeat myself.

Reply:
Thanks for pointing this out! As you mentioned, this was mostly used as scaffolding for writing the code itself, but I will remove in final submissions moving forward! :)